llvm/lib/SYCLLowerIR/LowerWGScope.cpp

```diff
@@ -630,6 +632,11 @@ static void fixupPrivateMemoryPFWILambdaCaptures(CallInst *PFWICall) {
   // whether it is an alloca with "work_item_scope"
   SmallVector<CaptureDesc, 4> PrivMemCaptures;
 
+  // Look through cast
+  auto *Cast = dyn_cast<AddrSpaceCastInst>(LambdaObj);
+  if (Cast)
```

Comment:
Combine into `if (auto *Cast = dyn_cast<AddrSpaceCastInst>(LambdaObj))`

Reply:
Done.

This is a cherry-pick from https://github.com/intel/llvm/pull/3120. I duplicated this comment there as well.